gcp/website/frontend_handlers.py

```diff
@@ -869,8 +867,22 @@ def print_subtree(vuln_id: str) -> None:
   return final_string
 
 
-def get_upstreams_of_vulnerability(target_bug_id: str,
-                                   transitive_upstreams: list[str]) -> str:
+def reverse_tree(graph: dict[str, set[str]]) -> dict[str, set[str]]:
+  """
+  Reverses a graph represented as a dictionary
+  """
+
+  reversed_graph = {}
+  for node, children in graph.items():
+    for child in children:
+      if child not in reversed_graph:
+        reversed_graph[child] = set()
+      reversed_graph[child].add(node)
+
+  return reversed_graph
+
+
+def get_upstreams_of_vulnerability(bug) -> str:
```

Comment:
nit: should the return type be `-> str | None`? I see it returns None when cycle detected